src/main/java/algorithms/medium/BestTimeToBuyAndSellStockWithCooldown.java

```diff
@@ -0,0 +1,24 @@
+package algorithms.medium;
+
+public class BestTimeToBuyAndSellStockWithCooldown {
+
+    public int maxProfit(int[] prices) {
+
+        int totalProfit = 0, restProfit = 0, currBuyCosts = Integer.MIN_VALUE;
```

Comment:
Is there a better naming for the variable `currBuyCosts`? It basically represents the state of our income when we have invested but have not earned from its returns.

Reply:
How about investmentCost? 
"Cost of Investment" is also used commonly in finance.

Reply:
It is called "`hold`" state in investment by the way.
Bitcoiners call it "`hodl`" as a joke, because of a Reddit post.